### PR TITLE
sql: add syntax for ALTER DATABASE ... SET sessionvar

### LINKED ISSUE #26947: sql: support database-level session variable defaults
Drupal does the following:

`ALTER DATABASE "drupal" SET bytea_output = 'escape';`

which is mean to set a database-level default setting for session vars.

See https://www.postgresql.org/docs/10/static/sql-alterdatabase.html:

> The remaining forms change the session default for a run-time configuration variable for a PostgreSQL database. Whenever a new session is subsequently started in that database, the specified value becomes the session default value. 

Also, as an addition to #21151, users can have database-level session variable defaults, with the `ALTER USER <username> IN DATABASE <database_name> SET ...` syntax. See https://www.postgresql.org/docs/current/sql-alteruser.html

The information is populated in pg_db_role_setting: https://www.postgresql.org/docs/current/catalog-pg-db-role-setting.html

Epic CRDB-2507

### PR DESCRIPTION
**Pull request by rafiss**
fixes #26947

See individual commits

Release note (sql change): The syntax for setting database placement is
now `ALTER DATABASE db PLACEMENT ...`. (The SET keyword no longer is
allowed before the PLACEMENT keyword.)

Release note (sql change): The `ALTER DATABASE db SET var ...` syntax is
now supported. It is a syntax alias for `ALTER ROLE ALL IN DATABASE db SET var ...`,
since it is identical to that functionality. That is, it
configures the default value to use for a session variable when a user
connects to the given database.

### CODE REVIEW

**Review by pawalt — COMMENTED** [review]
Placement changes LGTM except for the doc line.

> Reviewed 7 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69067#-:-MhPdipn--PDAcwlIuBQ:b-7bwdws)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[pkg/sql/parser/sql.y, line 1657 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69067#-MhPdLr64UtRNEGfRCHm:-MhPdLr64UtRNEGfRCHn:b-6b5rr2) ([raw file](https://github.com/cockroachdb/cockroach/blob/c304ea1991a63f7bac4a90aa0fa167a212bc6606/pkg/sql/parser/sql.y#L1657)):*
> ```Yacc
> // ALTER DATABASE <name> PRIMARY REGION <region>
> // ALTER DATABASE <name> SURVIVE <failure type>
> // ALTER DATABASE <name> PLACEMENT { RESTRICTED | DEFAULT }
> ```

Let's remove this doc line here. We're hiding PLACEMENT from users behind a cluster setting and only using it as a last-resort, so we don't want this comment leaking out.

File: pkg/sql/parser/testdata/alter_user [hunk 1]
```diff
@@ -301,3 +301,43 @@ ALTER ROLE ALL RESET ALL
...
+parse
+ALTER DATABASE d SET application_name = 'app'
+----
+ALTER ROLE ALL IN DATABASE d SET application_name = 'app' -- normalized!
+ALTER ROLE ALL IN DATABASE d SET application_name = ('app') -- fully parenthesized
+ALTER ROLE ALL IN DATABASE d SET application_name = _ -- literals removed
+ALTER ROLE ALL IN DATABASE _ SET application_name = 'app' -- identifiers removed
+
+parse
+ALTER DATABASE d RESET application_name
```
Comment (otan) [inline pkg/sql/parser/testdata/alter_user:314]: i thinky ou might need to rebase to catch #69053

**Review by otan — APPROVED** [review]

**Comment by awoods187** [comment]
This change makes sense to me--we dont have plans to widely use the placement syntax yet and if it enables the new functionality it seems worthwhile.

**Comment by rafiss** [comment]
tftr!

bors r=otan,pawalt

**Pull request merged**